test/parallel/test-buffer.js

```diff
@@ -1163,3 +1163,34 @@ assert.throws(function() {
   var b = new Buffer(1);
   b.equals('abc');
 });
+
+// Test indexOf
+var b = new Buffer('abcdef');
+
+assert.equal(b.indexOf('a'), 0);
+assert.equal(b.indexOf('a', 1), -1);
+assert.equal(b.indexOf('a', -1), 0);
```

Comment:
If we changed to `Array.prototype.indexOf`/`TypedArray.prototype.indexOf` semantics, this would change to:

```js
assert.equal(b.indexOf('a', -1), -1);
```

And I'd add a few more tests for good measure:

```js
assert.equal(b.indexOf('a', -4), -1);
assert.equal(b.indexOf('a', -5), 0);
```